doc/design/traingjob_crd_design.md

```diff
@@ -0,0 +1,172 @@
+# Design Doc for PaddlePaddle TrainingJob k8s CRD
+# Goals
+We are going to define PaddlePaddle cluster job using Kubernetes [CRD](https://kubernetes.io/docs/concepts/api-extension/custom-resources/). Each job is described using a yaml representation called  
+TrainingJob. With CRD, we can manage the PaddlePaddle cluster job as easy as k8s primary resources. The goals including:
+
+- Implement a controller and register a TrainingJob CRD
+- Create and delete TrainingJob
+- TrainingJob life cycle management
+
+# Requirements
+
+K8s 1.8+  and [Golang 1.9+](https://golang.org/dl/) are required for this project.
+
+# Design
+## concepts
+### TrainingJob
+The TrainingJob CRD defines a PaddlePaddle job resource for k8s, including cluster training parameters and a set of 
+PaddlePaddle roles such as master, pserver and trainer.
+
+- master is a k8s `ReplicaSet`
+- pserver is a k8s `ReplicaSet`
+- trainer is a k8s `job`
+
+Here is an example yaml for a PaddlePaddle cluster job.
+ 
+```yaml
+apiVersion: paddlepaddle.org/v1alpha1
+kind: TrainingJob
+metadata:
+  name: paddlejob
+  namespace: testspace
+spec:
+  #paddle image with training script
+  image: "paddlepaddle/paddlecloud-job"
+  # base port of pserver
+  port: 7164
+  # ports num default 1
```

Comment:
These comments seem not necessary, we can explain the details of the yaml file in another markdown file.